prusti-tests/tests/verify/fail/ghost-constraints/invalid-trait-refinement-1.rs

```diff
@@ -0,0 +1,47 @@
+use prusti_contracts::*;
+
+trait HasContract {
+    #[pure]
+    fn pre(&self) -> bool;
+    #[pure]
+    fn post(&self) -> bool;
+}
+
+trait MyTrait {
+    fn foo(&mut self);
+}
+
+#[extern_spec] //~ ERROR: the method's precondition may not be a valid weakening of the trait's precondition
```

Comment:
Why is it reported here?

Reply:
The full error message is:
```
error: [Prusti: verification error] the method's precondition may not be a valid weakening of the trait's precondition.
  --> verify-me.rs:15:1
   |
15 | #[extern_spec] //~ ERROR: the method's precondition may not be a valid weakening of the trait's precondition
   | ^^^^^^^^^^^^^^
...
29 |     #[requires(self.x >= 15)]
   |                ^^^^^^^^^^^^
   |
   = help: The trait's precondition should imply the implemented method's precondition.
   = note: this error originates in the attribute macro `extern_spec` (in Nightly builds, run with -Z macro-backtrace for more info)
```
which is fine imo.
The `#[extern_spec]` is part of the error span because the precondition function is generated during the expansion of the extern spec.
I will create UI tests for `invalid-trait-refinement-1.rs` and `invalid-trait-refinement-2.rs` to resolve this confusion (afaik it's not possible to test a multiline span nicely with these error patterns).